src/main/java/digital/slovensko/autogram/model/ProtectedDSSDocument.java

```diff
@@ -0,0 +1,12 @@
+package digital.slovensko.autogram.model;
+
+import eu.europa.esig.dss.model.DSSDocument;
+
+public interface ProtectedDSSDocument extends DSSDocument {
```

Comment:
Podľa mňa nám stačí jedna classa ProtectedDSSDocument, ktorá bude mať v sebe `DSSDocument` - to je interface nad InMemoryDocument aj FileDocument. A ešte by som to možno premenoval na niečo generickejšie - `AutogramDocument`?

Reply:
To by šlo, ale nebudeme mať priamo metódy od dss document a budeme musieť robiť nieco ako `AutogramDocument#getDocument`

Reply:
To prežijeme. Prípadne vieme implementovať tie, ktoré aj opakovane používame - zase až tak veľa ich nebude.